blazar/api/v1/oshosts/v1_0.py

```diff
@@ -79,3 +79,10 @@ def allocations_list(query, detail=False):
 def allocations_get(host_id, query):
     """List all allocations on a specific host."""
     return api_utils.render(allocation=_api.get_allocations(host_id, query))
+
+
+@rest.put('/<host_id>/allocation')
+@validation.check_exists(_api.get_computehost, host_id='host_id')
+def reallocate(host_id, data):
```

Comment:
What is `data` here? I think that's too general of a name. It's not clear what this argument is used for.

Reply:
I was under the impression we have to use data because of how the @rest decorator works.

```
if flask.request.method in ['POST', 'PUT']:
                    kwargs['data'] = request_data()
```

For put and post methods, it parses json in the request_data() function and then passes it as parameter data. I haven't tested this out so I could be mistaken.